docs/frontend.asciidoc

```diff
@@ -0,0 +1,108 @@
+[[frontend]]
+== Frontend support
+
+NOTE: Frontend support is an experimental feature not intended for production usage at the moment. 
+
+This section describes features specifically designed to support frontend application monitoring
+and a guide on how to enable experimental frontend support. 
+
+[[frontend-enable]]
+[float]
+=== Enable Frontend Support
+To try out experimental frontend support, set the `apm-server.frontend.enabled` to `true`.
+See https://github.com/elastic/apm-server/blob/{doc-branch}/apm-server.reference.yml[`apm-server.reference.yml`] for more configuration options.
+
+[[sourcemap]]
+=== Source Maps
+It is common practice to minify client-side JavaScript code for several reasons, e.g. performance gain. 
+This can make debugging very difficult, as it is hard to read the minified files.
+Source mapping can help for debugging minfied JavaScript files, 
+by mapping code from the minified files to the original source code. 
+
+APM Server has a source map <<sourcemap-endpoint,upload endpoint>> 
+which accepts source maps complying to the 
+https://docs.google.com/document/d/1U1RGAehQwRypUTovF1KRlpiOFze0b-_2gc6fAH0KY0k[Source map revision 3 proposal].
+
+Uploaded source maps are used to map `stack trace` information from recorded `transaction` and `error` documents 
+to the original source code files for easier debugging.  
+
+[[sourcemap-apply]]
+[float]
+==== How source maps are applied
+
+When source maps have been <<sourcemap-endpoint,uploaded>> and <<frontend-enable,frontend support is enabled>>, 
+source mapping is automatically applied to the `stack trace frames` of all `errors` and `transactions` 
+recorded with the https://github.com/elastic/apm-agent-js-base[JavaScript frontend agent].
+
+The server tries to find an uploaded source map for every `stack trace frame` of the record.
```

Comment:
With "try to find an uploaded source map" you are referring to some piece of documentation that the user didn't read yet, which might be confusing or force users to jump back and forth. 

my 2c: I'd put the `Upload Endpoint` section before the `How source maps are applied` section, and rename it to something like `Upload source maps`

Reply:
I first had it the other way around, but as the section `How source maps are applied` gives more overview about source mapping I thought it might be easier to understand to put this first.

The upload endpoint is linked in the first sentence of this section, and it is also referred to in the general description, that is why I don't link to it again:

> When source maps have been <<sourcemap-endpoint,uploaded>> and <<frontend-enable,frontend support is enabled>>, 

In case we change the order, the upload example which is rather comprehensive, is shown before the `How source maps are applied` section, so people might overlook this section.